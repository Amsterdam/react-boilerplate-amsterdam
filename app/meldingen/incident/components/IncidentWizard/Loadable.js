/**
 *
 * Asynchronously loads the component for IncidentWizard
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
