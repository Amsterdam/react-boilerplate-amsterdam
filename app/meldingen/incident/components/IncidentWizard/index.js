/**
*
* IncidentWizard
*
*/

import React from 'react';
// import PropTypes from 'prop-types';
import { BrowserRouter, Route } from 'react-router-dom';
import { Wizard, Steps, Step } from 'react-albus';

// import { FormattedMessage } from 'react-intl';
// import messages from './messages';

import wizard from '../../definitions/wizard';

import IncidentStep from '../IncidentStep';
import IncidentNavigation from '../IncidentNavigation';
import './style.scss';

function IncidentWizard() {
  Object.keys(wizard).map((key) => {
    console.log('key', key, wizard[key]);
    return true;
  });
  return (
    <BrowserRouter>
      <div className="incident-wizard">
        <div className="row pad-t">
          <div className="col-xs-6 col-xs-offset-3">
            <Route
              render={({ history }) => (
                <Wizard history={history}>
                  <Steps>
                    {Object.keys(wizard).map((key) => (
                      <Step key={key} id={`incident/${key}`}>
                        <h1 className="text-align-center">{key}</h1>
                        <IncidentStep content={wizard[key]} />
                      </Step>
                    )
                    )}
                  </Steps>
                  <IncidentNavigation />
                </Wizard>
              )}
            />
          </div>
        </div>
      </div>
    </BrowserRouter>
  );
}

// IncidentWizard.propTypes = {
// };

export default IncidentWizard;
