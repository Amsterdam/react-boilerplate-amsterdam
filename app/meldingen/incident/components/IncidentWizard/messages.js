/*
 * IncidentWizard Messages
 *
 * This contains all the text for the IncidentWizard component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.IncidentWizard.header',
    defaultMessage: 'This is the IncidentWizard component !',
  },
});
