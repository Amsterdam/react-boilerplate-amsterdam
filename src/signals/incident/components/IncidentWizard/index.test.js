import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import IncidentWizard from './index';

jest.mock('react-albus');
jest.mock('../IncidentForm', () => 'IncidentForm');
jest.mock('../IncidentPreview', () => 'IncidentPreview');

describe('<IncidentWizard />', () => {
  let props;

  beforeEach(() => {
    props = {
      wizardDefinition: {
        step1: {
          form: {}
        },
        step2: {
          form: {}
        },
        step3: {
          form: {},
          preview: {}
        },
        step4: {}
      },
      getClassification: jest.fn(),
      updateIncident: jest.fn(),
      createIncident: jest.fn(),
      incidentContainer: {},
      isAuthenticated: false
    };
  });

  it('expect to render correctly', () => {
    const wrapper = mount(
      <MemoryRouter
        initialEntries={[{
          pathName: 'incident/step1',
          key: 'wizard'
        }]}
      >
        <IncidentWizard {...props} />
      </MemoryRouter>);

    // expect(wrapper.find('Route')).toBe();
    expect(wrapper).toMatchSnapshot();
  });
});
