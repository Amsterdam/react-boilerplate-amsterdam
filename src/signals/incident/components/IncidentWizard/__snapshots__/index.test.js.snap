// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncidentWizard /> expect to render correctly 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "wizard",
        "pathName": "incident/step1",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "wizard",
            "pathName": "incident/step1",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "wizard",
          "pathName": "incident/step1",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <IncidentWizard
      createIncident={[MockFunction]}
      getClassification={[MockFunction]}
      incidentContainer={Object {}}
      isAuthenticated={false}
      updateIncident={[MockFunction]}
      wizardDefinition={
        Object {
          "step1": Object {
            "form": Object {},
          },
          "step2": Object {
            "form": Object {},
          },
          "step3": Object {
            "form": Object {},
            "preview": Object {},
          },
          "step4": Object {},
        }
      }
    >
      <div
        className="incident-wizard"
      >
        <Route
          render={[Function]}
        >
          <Wizard
            basename=""
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "wizard",
                    "pathName": "incident/step1",
                    "pathname": "/",
                    "search": "",
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "wizard",
                  "pathName": "incident/step1",
                  "pathname": "/",
                  "search": "",
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            onNext={[Function]}
            render={null}
          />
        </Route>
      </div>
    </IncidentWizard>
  </Router>
</MemoryRouter>
`;
